tests/manage/z_cluster/nodes/test_mds_running_on_different_nodes_post_shutdown_recovery.py

```diff
@@ -0,0 +1,139 @@
+from ocs_ci.ocs.resources.pod import get_all_pods
+from tests.helpers import get_node_running_pod
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@workloads
+class TestOCSWorkerNodeShutdown(E2ETest):
+    """
+    Test case validate both the MDS pods not running on same node
+    post shutdown and recovery
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, nodes):
+        """
+        Restart nodes that are in status NotReady
+        for situations in which the test failed in between
+
+        """
+
+        def finalizer():
+
+            # Validate all nodes are in READY state
+            not_ready_nodes = [
+                n for n in get_node_objs() if n
+                .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+            ]
+            log.warning(
+                f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+            )
+            if not_ready_nodes:
+                nodes.restart_nodes_by_stop_and_start(not_ready_nodes)
+                wait_for_nodes_status()
+
+            log.info("All nodes are in Ready status")
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.polarion_id("OCS-1278")
+    def test_mds_running_on_different_node_after_shutdown_and_recovery_worker_node(self, nodes):
+        """
+        Test case to check MDS pods not running on same node
+        post shutdown and recovery node
+
+        """
+
+        # Get MDS pods running nodes before shutdown
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE, selector=['rook-ceph-mds']
+        )
+        mds_pods_running_nodes_before_shutdown = []
+        for mds in pod_obj_list:
+            mds_pods_running_nodes_before_shutdown.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=mds.name
+                )
+            )
+        log.info(f"Mds running on nodes before shutdown {mds_pods_running_nodes_before_shutdown}")
+
+        # Get the node list
+        node = get_typed_nodes(node_type='worker', num_of_nodes=2)
+
+        # Shutdown 2 worker nodes for 10 mins
+        nodes.stop_nodes(nodes=node)
+
+        waiting_time = 600
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+
+        nodes.start_nodes(nodes=node)
+
+        # Validate all nodes are in READY state and up
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=30,
+            delay=15)(
+            wait_for_nodes_status(timeout=1800)
+        )
+
+        # Check the node are Ready state and check cluster is health ok
+        self.sanity_helpers.health_check()
```

Comment:
health check at the  end of the test as well would be good

Reply:
I think checking status after nodes recovery is good, After that its just assert statements so i feel the current one is okay, Let me know if you have any concerns